### ISSUE TITLE
[upstream issue] missing license and permission to distribute fissix/pgen2/tokenize.py

### ISSUE THREAD

**Issue by sten0** [jreese/fissix#8]
Hi,

I've begun packaging Fissix for Debian, as a dependency of Bowler, and as you may know Debian is very strict about licenses.  Unfortunately the following makes Fissix undistribuable:
```
# Copyright (c) 2001, 2002, 2003, 2004, 2005, 2006 Python Software Foundation.                        
# All rights reserved.
```
because a license grant permitting distribution is missing, and because redistribution is explicitly prohibited by "All rights reserved".  Eg: the right to redistribute has been reserved by the Python Software Foundation.

Would you please work with upstream lib2to3 to resolve this issue?  An additional line with "License: PSF" will solve it.

Thanks!
Nicholas

**Comment by jreese** [comment]
Unfortunately, upstream lib2to3 is now essentially unmaintained, so I don't expect the upstream license content to be updated.  It would be interesting to see how this was resolved for Parso or Redbaron, since those are both forks of lib2to3 - are those packages in Debian by chance? Bowler is planning to move from lib2to3/fissix to using [LibCST](https://github.com/Instagram/LibCST) instead, which is also based on Parso, so I'd personally like to know if Parso meets the Debian guidelines, as that will eventually be required instead of fissix.

**Comment by sten0** [comment]
> Unfortunately, upstream lib2to3 is now essentially unmaintained, so I
> don't expect the upstream license content to be updated.

Thanks for the news, I wasn't yet aware of this!

> It would be interesting to see how this was resolved for Parso or
> Redbaron, since those are both forks of lib2to3 - are those packages
> in Debian by chance?

Parso is, but Redbaron isn't.  Parso documents the lib2to3 copyright
like this, which is acceptable (documentation in the most upstream of
upstreams is the ideal):
  https://github.com/davidhalter/parso/blob/master/LICENSE.txt

Ideally Parso should provide a URL to the committish they forked or
merged from, so anyone can verify the claims in LICENSE.txt.

> Bowler is planning to move from lib2to3/fissix to using
> [LibCST](https://github.com/Instagram/LibCST) instead, which is also
> based on Parso, so I'd personally like to know if Parso meets the
> Debian guidelines, as that will eventually be required instead of
> fissix.

Parso 0.1.0 met an ftpmaster's evaluation of the standards for inclusion
in 2017.  I am optimistic that fissix would be accepted in 2020 with a
method similar to Parso's LICENSE.txt.  Periodic merges from lib2to3
that aren't represented in fissix's git history could be documented in
release notes, and then the combination of LICENSE.txt and reference to
a upstream lib2to3 committish can be used to evaluate the effective
license of the file in question.

Thank you for sharing this info about the migration to LibCST!  Do you
know what the timeline is for this?  If Fissix will be maintained
long-term and is useful for other projects, then I'd be happy to upload
it now, but please let me know.

>
Best,
Nicholas

**Comment by graingert** [comment]
@sten0 Just to let you know, I released python-modernize 0.8rc1 with a dep on fissix so fissix will need to be packages for that too
